services/DeviceInformation/include/ble-service-device-information/DeviceInformationService.h

```diff
@@ -0,0 +1,92 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_DEVICE_INFORMATION_SERVICE_H
+#define BLE_DEVICE_INFORMATION_SERVICE_H
+
+#include "ble/BLE.h"
+
+#if BLE_FEATURE_GATT_SERVER
+
+/** BLE Device Information Service class doesn't require instantiation - the characteristics added are read only.
```

Comment:
I'm not sure I understand the notice, does it means that characteristics values are immutable on the server side ? 
If the service is immutable we can probably `delete` the constructor and destructor as well as the copy constructor and asignment operators. 
I'd also expect some background explaining what's the service is for.

Reply:
`I'm not sure I understand the notice, does it means that characteristics values are immutable on the server side ?`
Yes, the characteristic is written only once, presumably at boot. Information here can only change between boots.

I'll remove the constructor.